integration-tests/tests/circuits.rs

```diff
@@ -100,33 +237,49 @@ async fn test_tx_circuit_block(block_num: u64) {
 
     let prover = MockProver::run(DEGREE, &circuit, vec![vec![]]).unwrap();
 
-    prover.verify_par().expect("tx_circuit verification failed");
+    prover.verify().expect("tx_circuit verification failed");
 }
 
 pub async fn test_bytecode_circuit_block(block_num: u64) {
     const DEGREE: u32 = 16;
 
+    let rng = XorShiftRng::from_seed([
+        0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc,
+        0xe5,
+    ]);
+
     log::info!("test bytecode circuit, block number: {}", block_num);
-    let cli = get_client();
-    let cli = BuilderClient::new(cli, CIRCUITS_PARAMS).await.unwrap();
-    let (builder, _) = cli.gen_inputs(block_num).await.unwrap();
+    let (builder, _) = gen_inputs(block_num).await;
+
     let bytecodes: Vec<Vec<u8>> = builder.code_db.0.values().cloned().collect();
+    let unrolled: Vec<_> = bytecodes.iter().map(|b| unroll::<Fr>(b.clone())).collect();
+
+    let circuit = BytecodeCircuitTester::<Fr>::new(unrolled, 2usize.pow(DEGREE));
 
-    test_bytecode_circuit::<Fr>(DEGREE, bytecodes);
+    test_run(rng, DEGREE, circuit, Vec::new());
 }
 
 pub async fn test_copy_circuit_block(block_num: u64) {
     const DEGREE: u32 = 16;
 
+    let rng = XorShiftRng::from_seed([
+        0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc,
+        0xe5,
+    ]);
+
     log::info!("test copy circuit, block number: {}", block_num);
-    let cli = get_client();
-    let cli = BuilderClient::new(cli, CIRCUITS_PARAMS).await.unwrap();
-    let (builder, _) = cli.gen_inputs(block_num).await.unwrap();
+    let (builder, _) = gen_inputs(block_num).await;
     let block = block_convert(&builder.block, &builder.code_db);
 
-    assert!(test_copy_circuit(DEGREE, block).is_ok());
-}
+    let randomness = CopyCircuitTester::<Fr>::get_randomness();
+    let circuit = CopyCircuitTester::<Fr>::new(block, randomness);
+
+    let num_rows = 1 << DEGREE;
+    const NUM_BLINDING_ROWS: usize = 7 - 1;
+    let instance = vec![vec![randomness; num_rows - NUM_BLINDING_ROWS]];
 
+    test_run(rng, DEGREE, circuit, instance);
+}
 pub async fn test_super_circuit_block(block_num: u64) {
```

Comment:
Can you add a comment with a `TODO` for the super circuit to add a test with the real prover?

Reply:
ok